src/main/java/com/googleintern/wfm/ruleengine/model/RuleValidationReport.java

```diff
@@ -133,6 +162,12 @@ public void writeToCsvFile(String outputCsvFilePath) throws IOException {
     };
   }
 
+  private ImmutableList<String[]> convertRulesToCsvRows() {
+    ImmutableList.Builder<String[]> rulesCsvRowsBuilder = ImmutableList.builder();
+    generatedRules().forEach(rule -> rulesCsvRowsBuilder.add(rule.toCSVRows()));
+    return rulesCsvRowsBuilder.build();
```

Comment:
This can be represented as a single streamy. 
generatedRules().stream().toImmutableList(Rule::toCSVRows);

and you can put it back if it is one line. 

Same elsewhere.